utils.go

```diff
@@ -16,18 +17,28 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a PROPAGATION_FORMAT_GO_HTTP_HEADER injector (our preference).
+	injector := sp.PropagationInjectorForFormat(PROPAGATION_FORMAT_GO_HTTP_HEADER)
```

Comment:
Code example to above comment.
(I don't know go so excuse me if it turns into psuedo)

```
// HttpHeaderFrameworkInjector
func Inject(
	sp Span,
	h http.Header,
) error {
	map<string,string> headerMap = h
	splitTextInjector.InjectSpan(sp, headerMap)
}

// SplitTextFormatInjector
func Inject(
	sp Span,
	map Map<string,string>,
) error {
	for headerSuffix, val := [span.traceId, span.spanId] {
		h.Add(ContextIDHTTPHeaderPrefix+headerSuffix, url.QueryEscape(val))
	}
	for headerSuffix, val := range span.TraceAttributes {
		h.Add(TagsHTTPHeaderPrefix+headerSuffix, url.QueryEscape(val))
	}
}
```